### DISCUSSION TITLE
How do I use supress_ipopt_output?

### DISCUSSION THREAD

**Discussion by DaDudeIan** [do-mpc/do-mpc#362] [Q&A] [answered]
Hello.

Basically just the question in the title. I'm having a difficult time debugging my code as I'm not getting what I'm expecting. The ipopt printout isn't exactly helping. I have my MPC declared like this:
```python
class BaseMPC:
  ...
  def setup_mpc(self)
    from do_mpc.model import Model
    from do_mpc.controller import MPC
    ...
    self.model = Model(model_type)
    # setup
    self.model.setup()
    ...
    self.mpc = MPC(self.model)
    # setup
    self.mpc.setup()
```
And when I run `self.mpc.make_step(x0)` it prints out these massive chunks to the terminal that I find confusing more than anything. Can someone please tell me how to use the `supress_ipopt_output` function from [here](https://www.do-mpc.com/en/develop/api/do_mpc.controller.MPCSettings.supress_ipopt_output.html#do_mpc.controller.MPCSettings.supress_ipopt_output), which comes from [MPCSettings](https://www.do-mpc.com/en/develop/api/do_mpc.controller.MPCSettings.html). I believe it's supposed to be called somewhere above `self.mpc.setup()`, but I just can't figure it out.

I really want this massive wall of prints to be blocked so it's only my own print statements that print anything to the terminal. Thank you!

**Comment by fralzo** [comment, marked as answer]
Hi, 
check #146
